app/Commands/Dev/DevCompile/NativeRust.hs

```diff
@@ -0,0 +1,75 @@
+module Commands.Dev.DevCompile.NativeRust where
+
+import Commands.Base
+import Commands.Dev.DevCompile.NativeRust.Options
+import Commands.Extra.Rust
+import Data.ByteString qualified as BS
+import Data.FileEmbed qualified as FE
+import Juvix.Compiler.Backend.Rust.Data.Result
+
+runCommand ::
+  (Members '[App, EmbedIO, TaggedLock] r) =>
+  NativeRustOptions 'InputMain ->
+  Sem r ()
+runCommand opts = do
+  let opts' = opts ^. nativeRustCompileCommonOptions
+      inputFile = opts' ^. compileInputFile
+      moutputFile = opts' ^. compileOutputFile
+  mainFile <- getMainFile inputFile
+  Result {..} <- runPipeline opts inputFile upToRust
+  rustFile <- inputRustFile mainFile
+  writeFileEnsureLn rustFile _resultRustCode
+  buildDir <- askBuildDir
+  ensureDir buildDir
+  prepareRuntime
+  outputFile <- nativeOutputFile mainFile moutputFile
+  let args =
+        RustArgs
+          { _rustDebug = opts' ^. compileDebug,
+            _rustInputFile = rustFile,
+            _rustOutputFile = outputFile,
+            _rustOptimizationLevel = fmap (min 3 . (+ 1)) (opts' ^. compileOptimizationLevel)
+          }
+  rustCompile args
+  where
+    prepareRuntime ::
+      forall s.
+      (Members '[App, EmbedIO] s) =>
+      Sem s ()
+    prepareRuntime = writeRuntime runtime
+      where
+        runtime :: BS.ByteString
+        runtime
+          | opts ^. nativeRustCompileCommonOptions . compileDebug = rustDebugRuntime
+          | otherwise = rustReleaseRuntime
+          where
+            rustReleaseRuntime :: BS.ByteString
+            rustReleaseRuntime = $(FE.makeRelativeToProject "runtime/rust/target/release/libjuvix.rlib" >>= FE.embedFile)
```

Comment:
I think we need an entry in the project package.yaml extra files for `runtime/rust` files.